Add release notes for v1.0.0 (#2139)
VirtusLab/scala-cli: website/docs/release_notes.md

# Release notes

## [v1.0.0](https://github.com/VirtusLab/scala-cli/releases/tag/v1.0.0)

## What's Changed

### Toolkit-test
By incorporating the [Scala Toolkit](https://github.com/scala/toolkit) into your project, you gain the advantage of two additional
dependencies seamlessly integrated into your classpath:
- `org.scala-lang:toolkit:<version>` is added to the main scope, allowing its utilization throughout your project.
- `org.scala-lang:toolkit-test:<version>` is included in the test scope, making it available exclusively for testing purposes.

> Scala CLI now supports this feature of Toolkit:

Reviewer: ```suggestion
Scala CLI now supports the following features for the toolkit:
```

* including e.g. `//> using toolkit latest` in any main scope file will automatically add the `toolkit` dependency to the main scope and the `toolkit-test` dependency to the test scope
* if you place e.g. `//> using toolkit latest` within a test scope file, both `toolkit` and `toolkit-test` will be limited to the test scope only

> * inserting e.g. `//> using test.toolkit latest` anywhere in the project will exclusively add `toolkit-test` to the test scope

Reviewer: ```suggestion
* inserting e.g. `//> using test.toolkit latest` anywhere in the project wil add both `toolkit` and `toolkit-test` to the test scope only
```

* Fix flaky completions for `zsh` by [@Jasper-M](https://github.com/Jasper-M) in [#2118](https://github.com/VirtusLab/scala-cli/pull/2118)
* Fix - install certificates for java by [@lwronski](https://github.com/lwronski) in [#2123](https://github.com/VirtusLab/scala-cli/pull/2123)
* Fix the `--source-jar` option & add corresponding `using` directives by [@Gedochao](https://github.com/Gedochao) in [#2120](https://github.com/VirtusLab/scala-cli/pull/2120)

### Documentation changes
* Add docs for bootstrapped standalone fat JAR by [@lwronski](https://github.com/lwronski) in [#2122](https://github.com/VirtusLab/scala-cli/pull/2122)
* Add developer docs on modifying `reflect-config.json` by [@Gedochao](https://github.com/Gedochao) in [#2114](https://github.com/VirtusLab/scala-cli/pull/2114)

### Build and internal changes
* Update release procedure - update also v1 tag by [@lwronski](https://github.com/lwronski) in [#2107](https://github.com/VirtusLab/scala-cli/pull/2107)
* NIT Refactor test scope directives by [@Gedochao](https://github.com/Gedochao) in [#2083](https://github.com/VirtusLab/scala-cli/pull/2083)

> * Add main class to jar manfiest in assembly by [@romanowski](https://github.com/romanowski) in [#2124](https://github.com/VirtusLab/scala-cli/pull/2124)

Reviewer: ```suggestion
* Add main class to jar manifest in assembly by [@romanowski](https://github.com/romanowski) in [#2124](https://github.com/VirtusLab/scala-cli/pull/2124)
```

dependencies seamlessly integrated into your classpath:
- `org.scala-lang:toolkit:<version>` is added to the main scope, allowing its utilization throughout your project.
- `org.scala-lang:toolkit-test:<version>` is included in the test scope, making it available exclusively for testing purposes.

Scala CLI now supports the following features for the toolkit:
* including e.g. `//> using toolkit latest` in any main scope file will automatically add the `toolkit` dependency to the main scope and the `toolkit-test` dependency to the test scope
* if you place e.g. `//> using toolkit latest` within a test scope file, both `toolkit` and `toolkit-test` will be limited to the test scope only
* inserting e.g. `//> using test.toolkit latest` anywhere in the project will add both `toolkit` and `toolkit-test` to the test scope only

This convention is encouraged for other toolkit-like libraries as well.

Added by [@Gedochao](https://github.com/Gedochao) in [#2127](https://github.com/VirtusLab/scala-cli/pull/2127) and [#2137](https://github.com/VirtusLab/scala-cli/pull/2137)

> ### Forcing script wrapper type

Reviewer: ```suggestion
### Forcing an object wrapper for scripts
```

> Scala CLI now supports the `//> using objectWrapper` directive, together with `--object-wrapper` option,
> which allows to force wrapping script code in an object body instead of a class.

Reviewer: ```suggestion
Scala CLI now supports the `//> using objectWrapper` directive, along with the corresponding `--object-wrapper` option,
which allows to force wrapping script code in an object body instead of a class.
```

> The object-wrapper should be avoided with multi-threading scripts, but may prove to be the only option in some cases.

Reviewer: ```suggestion
Using object wrappers should be avoided for scripts relying on multi-threading (as it may cause deadlocks), but may prove to be the only option in some cases.
```
